test/hotspot/jtreg/runtime/CDSCompressedKPtrs/XShareAuto.java

```diff
@@ -40,7 +40,7 @@
 public class XShareAuto {
     public static void main(String[] args) throws Exception {
         ProcessBuilder pb = ProcessTools.createLimitedTestJavaProcessBuilder(
-            "-server", "-XX:+UnlockDiagnosticVMOptions",
+            "-XX:+UnlockDiagnosticVMOptions",
```

Comment:
This test was created specifically as a regression test because `-server` didn't work. I'm not sure it serves any purpose now. @iklam?

Reply:
I think the test still serves as a functional test: if an archive is specified, then -Xshare:auto is used by default.

Reply:
Looking back at the history, the `-Xshare:auto` & `-server` issue (https://bugs.openjdk.org/browse/JDK-8005933) was related to this code in JDK 8:

https://github.com/openjdk/jdk8u-dev/commit/a1f3a95880318c32169ff89dc8784a0dfc629eec

In JDK 8, on c2 sharedSpaces is disabled if the following condition is true.

```
!DumpSharedSpaces && !RequireSharedSpaces &&
      (FLAG_IS_DEFAULT(UseSharedSpaces) || !UseSharedSpaces)
```

That code no longer exists in mainline. The `-server` option has become a no-op. The launcher code, `CheckJvmType` removes `-server` from the command line arguments passed to the VM side. As a result of those, `-server` is no real impact on CDS now.

Reply:
For this particular test, I think it's OK to remove `-server` (so it won't fail with static JDK) and ` * @bug 8005933`  (as the condition related to that particular bug no longer exists). But we should keep the test for the purpose "-Xshare:auto is the default when -Xshare is not specified"

Reply:
Thanks. I agree with keeping the test. That's also related to @AlanBateman comments on if the test needs to be skipped for static JDK.